docs/design.md

```diff
@@ -1,96 +0,0 @@
-## Design Document
```

Comment:
where did this go ?

Reply:
it's in the body of `sdk/overview.rst`, see http://cosmos-sdk.readthedocs.io/en/docs-refactor/sdk/overview.html